care/utils/assetintegration/onvif.py

```diff
@@ -0,0 +1,53 @@
+import enum
+import json
+from urllib import request
+
+from care.utils.assetintegration.base import BaseAssetIntegration
+
+
+class OnvifAsset(BaseAssetIntegration):
+    _name = "onvif"
+
+    class OnvifActions(enum.Enum):
+        MOVE_ABSOLUTE = "move_absolute"
+        GOTO_PRESET = "goto_preset"
+
+    def __init__(self, meta):
+        try:
+            self.meta = json.loads(meta)
+            self.meta = meta
+            self.name = meta["camera_type"]
+            self.host = meta["camera_address"]
+            self.port = meta["camera_port"] or 80
+            self.username = meta["camera_access_key"].split(":")[0]
+            self.password = meta["access_credentials"].split(":")[1]
+            self.middleware_hostname = meta["middleware_hostname"]
+        except KeyError:
+            print("Error: Invalid Onvif Asset")
+
+    def get_url(self, endpoint):
+        return "http://{}{}".format(self.middleware_hostname, endpoint)
+
+    def api_post(self, url, data=None):
+        req = request.post(url, json=data)
+
+    def api_get(self, url, data=None):
+        req = request.get(url, data=data)
+
+    def handle_action(self, action):
+        self.validate_camera(self)
```

Comment:
validate_camera does not seem to be a method in this class, no need to pass self explicitly since the first parameter is always self.

Reply:
Validation has been moved to __init__, so removing the call